core/vm/evm.go

```diff
@@ -144,15 +151,17 @@ type EVM struct {
 // NewEVM returns a new EVM. The returned EVM is not thread safe and should
 // only ever be used *once*.
 func NewEVM(blockCtx BlockContext, txCtx TxContext, statedb StateDB, chainConfig *params.ChainConfig, vmConfig Config) *EVM {
-	evm := &EVM{
-		Context:      blockCtx,
-		TxContext:    txCtx,
-		StateDB:      statedb,
-		vmConfig:     vmConfig,
-		chainConfig:  chainConfig,
-		chainRules:   chainConfig.Rules(blockCtx.BlockNumber),
-		interpreters: make([]Interpreter, 0, 1),
-	}
+	evm := EvmPool.Get().(*EVM)
```

Comment:
where do we put the evm back to pool

Reply:
![image](https://user-images.githubusercontent.com/7310198/126599066-91987b6e-fb80-4616-a331-04e4987b9015.png)